incubator/hnc/pkg/reconcilers/hnc_config_test.go

```diff
@@ -57,6 +57,7 @@ var _ = Describe("HNCConfiguration", func() {
 		Eventually(func() error {
 			return resetHNCConfigToDefault(ctx)
 		}).Should(Succeed())
+		cleanupObjects(ctx)
```

Comment:
I'm thinking that to be safe we should do it in `BeforeEach` AND `AfterEach`. It'd be a noop in `BeforeEach` if it's already empty. 

If we only do it in `BeforeEach`, the last test could in theory leave objects sitting on the server, only to be cleaned the next time a test suite is run.

Reply:
As for object_test.go, some of the tests do have `makeObject` in them, so wouldn't I need to make sure it's called still?

Reply:
Ok I added it in BeforeEach as well. Let me know what you think

Reply:
The apiserver is destroyed when the process exits so there's no danger there. You _can_ add this in object_test.go, but it's not really necessary since each object_test test runs in its own fresh set of namespaces, so there's no possibility of conflict. It's only because hnc-config looks across the _entire_ "cluster" that we get conflicts in those tests.